src/github_proxy.py

```diff
@@ -48,10 +50,19 @@ def publish_pr_comment(self, build):
 
         # initialize client before logging to ensure GitHub attributes are populated
         gh_client = self._get_client()
+        repo = gh_client.get_user(self._github_owner).get_repo(self._github_repo)
+
+        if config.DELETE_PREVIOUS_COMMENT == 'true':
+            comments = repo.get_issue(build.get_pr_id).get_comments()
+            for comment in comments:
+                if H3 in comment.body:
+                    LOG.debug('Deleting previous comment: repo=%s/%s, pr_id=%s, comment_id=%s',
+                              self._github_owner, self._github_repo, build.get_pr_id(), comment.id)
+                    comment.delete()
+                    break
```

Comment:
What order are the comments returned in? Will this delete the most recent or the oldest comment?

Reply:
Comments are ordered by ascending ID, so oldest first. You want to delete all matching comments in case it didn't succeed on the previous run?

Reply:
Yeah or if someone upgrades the app to pickup this new feature. It would clean up existing PRs.